src/rt/arrayassign.d

```diff
@@ -30,8 +30,25 @@ extern (C) void[] _d_arrayassign(TypeInfo ti, void[] from, void[] to)
 
     // Need a temporary buffer tmp[] big enough to hold one element
     void[16] buf = void;
-    void* ptmp = (elementSize > buf.sizeof) ? alloca(elementSize) : buf.ptr;
-    return _d_arrayassign_l(ti, from, to, ptmp);
+    void* ptmp;
+    if (elementSize <= buf.sizeof)
+    {
+        ptmp = buf.ptr;
+    }
+    else if (elementSize < 1024)
```

Comment:
How was 1024 chosen?

Reply:
Pretty random. Should be enough for a lot of data types, and not too much.